README.md

```diff
@@ -1,3 +1,607 @@
-# @octokit/webhooks.js
+# @octokit/webhooks
 
-> GitHub webhooks toolset
+> GitHub webhooks toolset for Node.js
+
+[GitHub webhooks](https://developer.github.com/webhooks/) can be registered in multiple ways
+
+1. In repository or organization settings on [github.com](https://github.com/).
+2. Using the REST API for [repositories](https://developer.github.com/v3/repos/hooks/) or [organizations](https://developer.github.com/v3/orgs/hooks/)
+3. By installing a [GitHub App](https://developer.github.com/apps/).
+
+`@octokit/webhooks` helps you handle the webhooks sent by GitHub to your server.
+
+## Example
+
+```js
+// install with: npm install @octokit/webhooks
+const Webhooks = require('@octokit/webhooks')
+const receiver = Webhooks.receiver({secret: 'mysecret'})
+
+// called with installation webhook payload
+// see https://developer.github.com/v3/activity/events/types/#installationevent
+receiver.on('installation.deleted', {installation, sender} => {
+  console.log(`${sender.login} deleted installation for ${installation.account.login}`)
+  // logs e.g. "octocat deleted installation for octocat"
+})
+
+// called with installation event payload, same as above.
+// Hook handlers can return a promise or throw errors
+receiver.hook('installation.deleted', {installation} => {
```

Comment:
Does async support need to be a separate method? Could we make `on` behave properly if a promise is returned or an error is thrown?

Reply:
We could but the Node EventEmitter is used so widely, it would break a widely used convention, so I’d rather not do that. I think event handlers are still great and they are more performent. Making the asynchronous handlers with a dedicated method makes it clear.

Reply:
> the Node EventEmitter is used so widely, it would break a widely used convention

What would the consequences be for those that expecting this to behave exactly as an `EventEmitter`?  If they treat it as an `EventEmitter`, it should work exactly as they expect. The only difference being that if they return a promise or throw an exception, the `receiver.handle` (which already doesn't look like an `EventEmitter`) would be able to find out about it. For most people, they won't event be exposed to that because they'll be using the middleware that abstracts this.

We're already breaking conventions with `on` accepting an array and the emitter being called `handle` instead of `emit`. As for performance, I'd have to see the implementation, but I'm skeptical that there will be a difference.

> Making the asynchronous handlers with a dedicated method makes it clear.

Personally, I have a strong preference for libraries that just do the right thing with promises. Being forced to use a different method based on the return semantics of your method has a smell to me.

I'm fine with settling for two separate methods if we have to, but I would only do that if there was significant trade-offs to make `on` work with both sync and async handlers.

Reply:
The biggest risk I see is that people do something like this 

```
middleware.on('installation', async (payload) => {
  await doSomethingThatTakesTooLong(payload)
})
```

I would not expect my `.on` handler to be blocking the response to the GitHub webhook. That’s probably the only relevant downside that I see.

How about we get rid of the `.on` / `.removeListener` methods instead?

Reply:
🤔 

I forgot to implement an `"error"` event which is something we should do. And with an error event I would definitely want to use the `.on('error', handleError)` API. 

Now I’m tending towards just the `.on()` API as Brandon originally suggested 😁 We can my concern by documenting that it does _not_ behave like a EventEmitter event handler if the passed function is async or if the function returns a promise